scrapers/codechef.py

```diff
@@ -1,57 +1,117 @@
+        if contest_code[0:4] == "COOK":
+            # For cook-off contests, there's an extra penalty provision, we need only the alternate elements.
+            scores = scores[::2]            
+            ''' Extract all time elements, remove semi-colons, convert to numeric type. 
+                Smaller the time, the better. But score needs to be generated with left most digit being
+                number of points, hence time's subtracted from 100000 to allow descending order of scores '''            
+            times = list(100000-float(time.text.replace(":","")) for time in load_all(r'total-time', 'class'))
+            # The score's added to the left of the time             
+            scraped_scoreboard.extend(zip(
+                [x.find_elements_by_tag_name("span")[-1].text for x in names],
+                [float(scores[i].text.split()[0])*100000+times[i] for i in range(len(scores))]))        
+        else:
+            scraped_scoreboard.extend(zip(
+                [x.find_elements_by_tag_name("span")[-1].text for x in names],
+                [float(y.text.split()[0]) for y in scores]))                
+        if page == total_pages-1: # Reached Last Page
+            break    
+        driver.get(site + f'&page={page+2}') # go to next page
+
+def scrape(contest_codes):
+    scoreboard_base_url:str = "https://www.codechef.com/rankings"
+    site_url:str = "https://www.codechef.com"
+    scoreboard_filter_query:str = "?filterBy=Institution%3DPES%20University&itemsPerPage=100&order=asc&sortBy=rank"
+
+    for contest_code in contest_codes:
+        info(f'Codechef contest {contest_code}:')
+        final_scoreboard:list=list()
+        easy_points:int=100 # Points to add to division A participants assuming they can solve all easy div B problems
+
+        if contest_code[0:4]=="COOK":
+            easy_points=100000 # Initial value set to points per problem
+
+        for Division in divisions: # Build the scraped scoreboard
+            divisions[Division].problems = get_problems(f"{site_url}/{contest_code}{Division}")
+            divisions[Division].scraped_scoreboard = get_rankings(f"{scoreboard_base_url}/{contest_code}{Division}{scoreboard_filter_query}", contest_code)
+
+        easy_points=len(divisions['B'].problems-divisions['A'].problems)*easy_points # Points to add to div-A participants
+
+        for i in range(len(divisions['A'].scraped_scoreboard)): # Add easy points to all div-A participants
+            divisions['A'].scraped_scoreboard[i]=divisions['A'].scraped_scoreboard[i][0],divisions['A'].scraped_scoreboard[i][1]+easy_points
+            
+        final_scoreboard.extend(divisions['A'].scraped_scoreboard)
+        final_scoreboard.extend(divisions['B'].scraped_scoreboard) # Consolidate scoreboard
+        final_scoreboard = sorted(final_scoreboard,key=lambda x:x[1],reverse=True) # Sort list in desc. order based on points. (username, points)
+
+        if contest_code[0:5] == 'LTIME':
```

Comment:
This should probably be moved to the unified scrapers file too, let the only job of the scraper file be to fetch the rankings, offload everything else to the main file. The task of writing has already been offloaded to that, so I think this should too.

Reply:
this also is a codechef-specific thing; none of the other platforms require intricate renaming..

Reply:
okay; will do anyway